assets/chat/js/messages.js

```diff
@@ -33,11 +32,24 @@ function setFormattersFromSettings(settings) {
     if (settings.get('formatter-green')) formatters.set('green', new GreenTextFormatter());
 }
 
-function buildMessageTxt(chat, message){
+function buildMessageTxt(chat, message) {
     // TODO we strip off the `/me ` of every message -- must be a better way to do this
-    let msg = message.message.substring(0, 4).toLowerCase() === '/me ' ? message.message.substring(4) : message.message
-    formatters.forEach(f => msg = f.format(chat, msg, message))
-    return `<span class="text">${msg}</span>`
+    let msg = message.message.substring(0, 4).toLowerCase() === '/me ' ? message.message.substring(4) : message.message;
+    var codeFmt = new CodeFormatter();
+    var msgArray = codeFmt.split(msg);
+    var fullMsg = '';
+    for (var i = 0; i < msgArray.length; i++) {
+        if (msgArray[i][1] === '0') {
```

Comment:
can we please consider doing something like this:

```js
{
    type: 'code',
    value: 'some code text'
}

{
    type: 'text', // or any other "type" to represent _plain_ text
    value: 'normal, non-code text'
}
```

instead of

```js
// have to know that first element is value, second is type
[value, type]

// have to know that '1' is for code text
['some code text', '1']

// have to know that '0' is for non-code text
['normal, non-code text', '0']
```

I think this would be easier to read, and also easier to extend as new "types" are added. Instead of being another "number", we can simply give them a proper type name.